server/index.ts

```diff
@@ -1,122 +1,115 @@
+  ROOM_CREATED = 'roomCreated',
+  JOIN_ROOM = 'joinRoom',
+  JOINED_SUCCESSFULLY = 'joinedSuccessfully',
+  NEW_PLAYER = 'newPlayer',
+  START_ROOM = 'startRoom',
+  ROOM_STARTED = 'roomStarted',
+  PLAY = 'play',
+  MOVE_PLAYED = 'movePlayed',
+  PLAYER_DISCONNECTED = 'playerDisconnected',
 }
 
-io.on("connection", (socket) => {
+io.on('connection', socket => {
   console.log(`${socket.id} has connected!`);
-  socket.on(SocketEvent.CREATE_ROOM, async (data: { open: boolean }) => {
+  socket.on(SocketEvent.CREATE_ROOM, async ({ open }: { open: boolean }) => {
     try {
-      const room = await roomController.createRoom(data.open);
+      const room = await roomController.createRoom(open);
       socket.emit(SocketEvent.ROOM_CREATED, room);
     } catch (err) {
-      socket.emit("error", err.toString());
+      socket.emit('error', err.toString());
     }
   });
   socket.on(
     SocketEvent.JOIN_ROOM,
-    async (data: { roomCode: string; playerName: string }) => {
+    async ({
+      roomCode,
+      playerName,
+    }: {
+      roomCode: string;
+      playerName: string;
+    }) => {
       try {
         const { room, player } = await roomController.joinRoom(
-          data.roomCode,
+          roomCode,
           socket.id,
-          data.playerName
+          playerName
         );
         socket.join(room.roomCode);
         socket.broadcast.to(room.roomCode).emit(SocketEvent.NEW_PLAYER, player);
         socket.emit(SocketEvent.JOINED_SUCCESSFULLY, room);
       } catch (err) {
-        socket.emit("error", err.toString());
+        socket.emit('error', err.toString());
       }
     }
   );
 
   socket.on(SocketEvent.START_ROOM, async () => {
     try {
-      const room = await roomController.startGame(socket.id);
-      const { board, players } = room;
-      io.to(room.roomCode).emit(SocketEvent.ROOM_STARTED, { board, players });
+      const room = await roomController.startRoom(socket.id);
+      io.to(room.roomCode).emit(SocketEvent.ROOM_STARTED, {
+        board: room.board,
+        players: room.players,
+      });
     } catch (err) {
-      socket.emit("error", err.toString());
+      socket.emit('error', err.toString());
     }
   });
 
-  socket.on(SocketEvent.PLAY, async (data: { cards: Array<CardClass> }) => {
+  socket.on(SocketEvent.PLAY, async ({ cards }: { cards: Card[] }) => {
     try {
-      const { room, updated, player } = await roomController.playMove(
-        socket.id,
-        data.cards
-      );
-      io.to(room.roomCode).emit(SocketEvent.MOVE_PLAYED, {
-        room,
+      const {
+        name,
+        cards: playedCards,
         updated,
-        player,
+        board,
+        roomCode,
+      } = await roomController.playMove(socket.id, cards);
+      io.to(roomCode).emit(SocketEvent.MOVE_PLAYED, {
+        name,
+        cards: playedCards,
+        updated,
+        board,
       });
-    } catch (err) {}
+    } catch (err) {
+      socket.emit('eror', err.toString());
```

Comment:
Should probably be error for consistency

Reply:
Good catch. This was a typo.